mlir/lib/Ion/Transforms/oqd_database_managers.hpp

```diff
@@ -0,0 +1,146 @@
+// Copyright 2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <iostream>
+
+#include <cassert>
+#include <regex>
+
+#include <toml++/toml.hpp>
+
+#include "oqd_database_types.hpp"
+#include "oqd_database_utils.hpp"
+
+namespace {
+
+// TODO: we should decide where the calibration tomls will live.
+// Very likely, they will be dumped into some local locations that are not part of the catalyst
+// tree. It would be best to store the toml locations as pass options on the --quantum-to-ion pass
+// This way, they can be directly specified in the frontend python layer, and passed down
+// to the pass as pass options (via the standard transform dialect lowering pipeline we have)
+// But for now this hard coded prototype is good enough.
+
+static const std::string catalyst_root_path =
+    std::regex_replace(__FILE__, std::regex("mlir/(.)*/oqd_database_managers.hpp"), "");
+
+static const std::string oqd_device_parameters_toml_file_path =
+    catalyst_root_path + "frontend/catalyst/third_party/oqd/src/oqd_device_parameters.toml";
+static const std::string oqd_qubit_parameters_toml_file_path =
+    catalyst_root_path + "frontend/catalyst/third_party/oqd/src/oqd_qubit_parameters.toml";
+static const std::string oqd_gate_decomposition_parameters_toml_file_path =
+    catalyst_root_path +
+    "frontend/catalyst/third_party/oqd/src/oqd_gate_decomposition_parameters.toml";
```

Comment:
I like this idea, it's similar to what I had done in the frontend `OQDDatabaseManager`s. Eventually it would be nice if a user could pass the path to the directory containing the database files when they instantiate the oqd device, something like:

```python
dev = qml.device("oqd", backend="default", database_dir="/path/to/databases", shots=1024, wires=2)
```

but this will come later. For now, I think we can just create the mechanism to give this path directly to the quantum-to-ion pass, like you wrote in your comment.

Reply:
> I think we can just create the mechanism to give this path directly to the quantum-to-ion pass

I will do that when this is merged into the big ion dialect branch

Reply:
Sounds good!